sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BoundAttribute.scala

```diff
@@ -31,7 +31,7 @@ import org.apache.spark.sql.types._
 case class BoundReference(ordinal: Int, dataType: DataType, nullable: Boolean)
   extends LeafExpression with NamedExpression {
```

Comment:
Its kinda of a hack, but sometimes after transforms we end up with BoundReferences in the place of fields that were `AttributeReference` and so there were class cast exceptions.  We might be able to remove this some day or now?